src/test/java/com/commercetools/sync/inventory/InventorySyncTest.java

```diff
@@ -0,0 +1,214 @@
+        final InventoryEntryDraft draftWithNewChannel = InventoryEntryDraft.of(SKU_3, QUANTITY_1, DATE_1, RESTOCKABLE_1,
+                Channel.referenceOfId(KEY_3));
+        final InventorySync inventorySync = getInventorySyncer(30, 1, true);
+        inventorySync.syncDrafts(asList(draftWithNewChannel));
+        final InventorySyncStatistics stats = inventorySync.getStatistics();
+        assertThat(stats.getProcessed()).isEqualTo(1);
+        assertThat(stats.getCreated()).isEqualTo(1);
+    }
+
+    @Test
+    public void syncDrafts_notSyncEntriesWithUnknownChannels_havingNotEnsuredChannels() {
+        final InventoryEntryDraft draftWithNewChannel = InventoryEntryDraft.of(SKU_3, QUANTITY_1, DATE_1, RESTOCKABLE_1,
+                Channel.referenceOfId(KEY_3));
+        final InventorySync inventorySync = getInventorySyncer(30, 1, false);
+        inventorySync.syncDrafts(asList(draftWithNewChannel));
+        final InventorySyncStatistics stats = inventorySync.getStatistics();
+        assertThat(stats.getProcessed()).isEqualTo(1);
+        assertThat(stats.getFailed()).isEqualTo(1);
+    }
+
+    @Test
+    public void syncDrafts_notSyncEntriesWithNullSku() {
+        final InventoryEntryDraft draftWithNullSku = InventoryEntryDraft.of(null, 12);
+        final InventorySync inventorySync = getInventorySyncer(30, 1, false);
+        inventorySync.syncDrafts(asList(draftWithNullSku));
+        final InventorySyncStatistics stats = inventorySync.getStatistics();
+        assertThat(stats.getProcessed()).isEqualTo(0);
+        assertThat(stats.getUnprocessedDueToEmptySku()).isEqualTo(1);
+    }
+
+    @Test
+    public void syncDrafts_returnsWithoutException_havingProblemsEncountered() {
+        final InventoryEntryDraft draftWithNewChannel = InventoryEntryDraft.of(SKU_3, QUANTITY_1, DATE_1, RESTOCKABLE_1,
+                Channel.referenceOfId(KEY_3));
+        final List<InventoryEntryDraft> toProcess = new ArrayList<>(drafts);
+        toProcess.add(draftWithNewChannel);
+        final InventorySync inventorySync = new InventorySync(InventorySyncOptionsBuilder.of(mockCtpClient())
+                .build(), mockThrowingInventoryService(), mock(TypeService.class));
+
+        inventorySync.syncDrafts(toProcess);
+        final InventorySyncStatistics stats = inventorySync.getStatistics();
+        assertThat(stats.getProcessed()).isEqualTo(10);
+        assertThat(stats.getFailed()).isEqualTo(7);
+    }
+
+    private InventorySync getInventorySyncer(int batchSize, int parallelThreads, boolean ensureChannels) {
+        final InventorySyncOptions options = InventorySyncOptionsBuilder.of(mockCtpClient())
+                .setBatchSize(batchSize)
+                .setParallelProcessing(parallelThreads)
+                .ensureChannels(ensureChannels)
+                .build();
+        return new InventorySync(options, mockInventoryService(), mock(TypeService.class));
+    }
+
+    private CtpClient mockCtpClient() {
```

Comment:
this could be added in the MockUtils class in the commons because it can be used by several modules in their tests.